sections/shared/Layout/AppLayout/Header/WalletActions.tsx

```diff
@@ -0,0 +1,174 @@
+import { FC, useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import styled, { useTheme } from 'styled-components';
+import { FlexDivRow } from 'styles/common';
+
+import Connector from 'containers/Connector';
+import { truncatedWalletAddressState } from 'store/wallet';
+import { useRecoilValue } from 'recoil';
+
+import CaretDownIcon from 'assets/svg/app/caret-down.svg';
+import DisconnectIcon from 'assets/svg/app/disconnect.svg';
+import SwitchWalletIcon from 'assets/svg/app/switch-wallet.svg';
+import { Svg } from 'react-optimized-image';
+
+import { components } from 'react-select';
+import Select from 'components/Select';
+import { IndicatorSeparator } from 'components/Select/Select';
+
+import getENSName from './UserMenu/getENSName';
+import ConnectionDot from './ConnectionDot';
+
+type ReactSelectOptionProps = {
+	label: string;
+	postfixIcon?: string;
+	isMenuLabel?: boolean;
+	onClick?: () => {};
+};
+
+export const WalletActions: FC = () => {
+	const { t } = useTranslation();
+	const theme = useTheme();
+	const {
+		connectWallet,
+		disconnectWallet,
+		switchAccounts,
+		isHardwareWallet,
+		signer,
+		staticMainnetProvider,
+	} = Connector.useContainer();
+	const hardwareWallet = isHardwareWallet();
+
+	const [ensName, setEns] = useState<string>('');
+	const truncatedWalletAddress = useRecoilValue(truncatedWalletAddressState);
+
+	const walletLabel = ensName ? ensName : truncatedWalletAddress;
+
+	const WALLET_OPTIONS = useMemo(() => {
+		let options = [
+			{ label: 'common.wallet.switch-wallet', postfixIcon: 'Switch', onClick: connectWallet },
+			{
+				label: 'common.wallet.disconnect-wallet',
+				postfixIcon: 'Disconnet',
+				onClick: disconnectWallet,
+			},
+		];
+
+		if (hardwareWallet) {
+			options.push({
+				label: 'common.wallet.switch-accounts',
+				postfixIcon: 'Switch',
+				onClick: switchAccounts,
+			});
+		}
+
+		return options;
+	}, [hardwareWallet]);
+
+	const formatOptionLabel = ({
+		label,
+		isMenuLabel,
+		postfixIcon,
+		onClick,
+	}: ReactSelectOptionProps) =>
+		isMenuLabel ? (
+			<>
+				<StyledConnectionDot />
+				{label}
+			</>
+		) : (
+			<LabelContainer onClick={onClick}>
+				{t(label)}
+				{postfixIcon &&
+					(postfixIcon === 'Switch' ? (
+						<Svg src={SwitchWalletIcon} height={17} />
+					) : (
+						<Svg src={DisconnectIcon} height={17} />
+					))}
+			</LabelContainer>
+		);
+
+	const DropdownIndicator = (props: any) => {
+		return (
+			<components.DropdownIndicator {...props}>
+				<StyledCaretDownIcon
+					src={CaretDownIcon}
+					viewBox={`0 0 ${CaretDownIcon.width} ${CaretDownIcon.height}`}
+				/>
+			</components.DropdownIndicator>
+		);
+	};
+
+	useEffect(() => {
+		if (signer) {
+			signer.getAddress().then((account: string) => {
+				const _account = account;
+				getENSName(_account, staticMainnetProvider).then((_ensName: string) => {
+					setEns(_ensName);
+				});
+			});
+		}
+	}, [signer, truncatedWalletAddress]);
```

Comment:
moved this from `UserMenu` into the `WalletActions themselves`

I was noticing some wallet switching issues ENS names and regular wallets. When you have a wallet thats connected to ENS, then switch to a wallet that does not have an ENS, it was failing to update the wallet address. I narrowed this down to two issues:

1) there used to be a [conditional for `setEns(_ensName)`](https://github.com/Kwenta/kwenta/pull/583/files?diff=unified&w=0#diff-040621a7608ebb6a9f7ad541ee07248aab281c740eadac16d6d92189a246bc22L41) - removing this conditional allowed the ens name to be reset to null, allowing the `walletDisplay` value to update correctly.
2) the useEffect was ignoring the `truncatedWalletAddress` as change indicator - adding this to the dependencies retriggers the setEns and subsequentially the walletDisplay value.